src/transform/tests/test_runner.rs

```diff
@@ -48,27 +48,39 @@ mod tests {
 
     const TEST_GLOBAL_ID: GlobalId = GlobalId::Transient(1234567);
 
-    thread_local! {
-        static FULL_TRANSFORM_LIST: Vec<Box<dyn Transform>> = {
-            let features = OptimizerFeatures::default();
-            let typecheck_ctx = typecheck::empty_context();
```

Comment:
The testing bug was that this `typecheck_ctx` used to get shared among different test runs, so Typecheck runs from different test runs stepped on each other's toes through the expected types.

Reply:
(This test framework is on its way to be deprecated, so don't think too hard about this code.)